packages/hub/src/lib/list-datasets.ts

```diff
@@ -17,24 +44,35 @@ export interface DatasetEntry {
 	updatedAt: Date;
 }
 
-export async function* listDatasets(params?: {
+export async function* listDatasets<
+	const T extends Exclude<(typeof EXPANDABLE_KEYS)[number], (typeof EXPAND_KEYS)[number]> = never,
+>(params?: {
 	search?: {
 		owner?: string;
+		tags?: string[];
```

Comment:
or `filter` to stay closer to the actual param name?

Reply:
It's `filter` for the API, but what users see is more something like https://huggingface.co/models?language=ja&license=license:other&sort=trending where there's `language`, `library`, `license`, `other` etc.

(See original issue #497)

Since the API returns the `tags` field in the `listModels`, etc calls, to me it's more user-friendly to also request on `tags`.

Note that there are also other search params such as `search: {owner: ...}`. If we added `filter` it would be more general IMO, like `owner:xxx`, etc, like a query language